src/sst/core/baseComponent.h

```diff
@@ -605,22 +687,122 @@ class SubComponentSlotInfo {
+       @param args Arguments to be passed to constructor.  This
+       signature is defined in the API definition
+
+       For ease in backward compatibility to old API, this call will
+       try to load using new API and will fallback to old if
+       unsuccessful.
+    */
     template <typename T, class... ARGS>
-    void createAll(std::vector<T*>& vec, bool insertNulls, uint64_t share_flags, ARGS... args) const {
+    void createAll(std::vector<T*>& vec, uint64_t share_flags, ARGS... args) const {
         for ( int i = 0; i <= getMaxPopulatedSlotNumber(); ++i ) {
             T* sub = create<T>(i, share_flags, args...);
-            if ( sub != NULL || insertNulls ) vec.push_back(sub);
+            vec.push_back(sub);
         }
     }
 
+    /**
+       Create all user defined subcomponents (defined in input file to SST
+       run) for the slot.
+
+       @param vec Vector of pair<int,T*> that will hold the pointers
+       to the new subcomponents.  The int will hold the index from
+       which the subcomponent wass loaded.  Unoccupied indexes will be
+       skipped.  All components will be added to the end of the
+       vector.
+       @param share_flags Share flags to be used by subcomponent
+       @param args Arguments to be passed to constructor.  This
+       signature is defined in the API definition
+
+       For ease in backward compatibility to old API, this call will
+       try to load using new API and will fallback to old if
+       unsuccessful.
+    */
     template <typename T, class... ARGS>
-    void createAll(std::vector<T*>& vec, uint64_t share_flags, ARGS... args) const {
-        createAll<T,ARGS...>(vec,true,share_flags,args...);
+    void createAllSparse(std::vector<std::pair<int,T*> >& vec, uint64_t share_flags, ARGS... args) const {
+        for ( int i = 0; i <= getMaxPopulatedSlotNumber(); ++i ) {
+            T* sub = create<T>(i, share_flags, args...);
+            vec.push_back(i,sub);
+        }
+    }
+
+    /**
+       Create all user defined subcomponents (defined in input file to SST
+       run) for the slot.
+
+       @param vec Vector of T* that will hold the pointers
+       to the new subcomponents.  Unoccupied indexes will be
+       skipped.  All components will be added to the end of the
+       vector.
+       @param share_flags Share flags to be used by subcomponent
+       @param args Arguments to be passed to constructor.  This
+       signature is defined in the API definition
+
+       For ease in backward compatibility to old API, this call will
+       try to load using new API and will fallback to old if
+       unsuccessful.
+    */
+    template <typename T, class... ARGS>
+    void createAllSparse(std::vector<T*>& vec, uint64_t share_flags, ARGS... args) const {
+        for ( int i = 0; i <= getMaxPopulatedSlotNumber(); ++i ) {
+            T* sub = create<T>(i, share_flags, args...);
+            vec.push_back(sub);
```

Comment:
Should this check for sub == nullptr before push_back?

Reply:
Yes.  Cut too much from the previous code.